### ISSUE TITLE
doc: automatically link classnames to their documentation pages throughout API docs

### ISSUE THREAD

**Issue by onecrayon** [aurelia/framework#648]
**I'm submitting a feature request**

I'm not sure if this is the most appropriate place for this request, but I could not find the hub.html app that serves Aurelia documentation open sourced anywhere (so I can't try and implement this on my own to contribute).

Basically, anywhere there's a classname (or ideally, method/property/etc.) in the API docs served by hub.html, it would be brilliant if it was an `<a>` link to that class's documentation page. Right now, it's really difficult for someone not familiar with the framework to find the right page in the API docs, because I have to use the search tool just to figure out what package a class is contained in. If I could simply click the classname and go straight there, life would be much happier.

**Comment by EisenbergEffect** [comment]
Makes sense. We should be able to do that I think it's just a matter of going through all the docs and updating them.

**Comment by onecrayon** [comment]
The API docs are autogenerated by TypeDoc, which is why I was looking for the hub.html source (best I can tell from an outsider's perspective who has never worked with TypeDoc before, linking stuff up dynamically based on the TypeDoc-generated JSON is the most likely route to success; I thought about modifying the JSON directly as part of the Gulp `doc` step, but that seemed more prone to breakage since it adds an additional layer of complexity/parsing between the app displaying the info and TypeDoc's output).

In any case, thanks for considering this! It would make navigating the API docs a lot easier.
